source/extensions/filters/network/sni_dynamic_forward_proxy/proxy_filter.cc

```diff
@@ -33,14 +35,23 @@ Network::FilterStatus ProxyFilter::onNewConnection() {
     return Network::FilterStatus::Continue;
   }
 
-  // TODO(lizan): implement circuit breaker in SNI dynamic forward proxy like it is in HTTP:
-  // https://github.com/envoyproxy/envoy/blob/master/source/extensions/filters/http/dynamic_forward_proxy/proxy_filter.cc#L65
+  circuit_breaker_ = config_->cache().canCreateDnsRequest(absl::nullopt);
+
+  if (circuit_breaker_ == nullptr) {
+    ENVOY_CONN_LOG(debug, "pending request overflow", read_callbacks_->connection());
```

Comment:
should this reference dns overflow instead of request?

Reply:
Yes. I think that we should reference dns overflow.